docs/doc/60-contributing/03-rfcs/20220809-share.md

```diff
@@ -0,0 +1,171 @@
+grant_share_object_name := { DATABASE <db_name> | TABLE <db_name>.<table_name> }
+```
+
+### Revoke share object privilege FROM SHARE
+
+Revoke access to databases and other supported database objects (tables, views, etc.) in the share. Revoking permissions to these objects effectively removes them from the share, making them inaccessible to all tenants that have created databases in the share.
+
+Syntax:
+
+```sql
+REVOKE object_privilege ON grant_share_object_name FROM SHARE <share_name>
+
+object_privilege := [USAGE|SELECT|REFERENCE_USAGE]
+
+grant_share_object_name := { DATABASE <db_name> | TABLE <db_name>.<table_name> }
+```
+
+### Show Grants
+
+Lists all access control privileges that have been explicitly granted to tenants and shares.
+
+Syntax:
+
+```sql
+SHOW GRANTS
+SHOW GRANTS ON grant_share_object_name
+SHOW GRANTS TO {TENANT <tenant_name> | SHARE <share_name> }
+SHOW GRANTS OF SHARE <share_name>
+
+grant_share_object_name := { DATABASE <db_name> | TABLE <db_name>.<table_name> }
+```
+
+- `SHOW GRANTS`: Syntactically equivalent to SHOW GRANTS TO TENANT current_tenant. Lists all the roles granted to the current tenant.
+- `SHOW GRANTS ON grant_share_object_name`: Lists all privileges that have been granted on the object.
+- `SHOW GRANTS TO ...`:
+  - `TENANT <tenant_name>`: Lists all the privileges granted to the tenant.
+  - `SHARE <share_name>`: Lists all the privileges granted to the share.
+- `SHOW GRANTS OF SHARE <share_name>`: Lists all the tenants for the share and indicates the tenants that are using the share.
+
+### Create Database, Table FROM Share
+
+After tenants have permission to access the share objects, can create the database from the share.
+
+Syntax:
+
+```sql
+CREATE DATABASE <name> FROM SHARE <provider_tenant>.<share_name>
+```
+
+## Example of Using Share with SQL
+
+To create a share using SQL:
+
+1. Use the `CREATE SHARE` command to create an empty share.
+2. Use the `GRANT <privilege> … TO SHARE` command to add a database to the share and then selectively grant access to specific database objects (tables) to the share.
+3. Use the `ALTER SHARE` command to add one or more accounts access to the share.
+4. Use the `CREATE DATABASE ... FROM SHARE` command to create a database from the share.
+
+### Step1: Create an Empty Share
+
+The following example creates an empty share named t:
+
+```sql
+create share t;
+```
+
+### Step 2: Grant Privileges for a Database and Objects to the Share
+
+To include objects in a share, grant permissions to each object. When granting permissions, first grant usage rights to any container objects before granting usage rights to the objects in the container. For example, grant access to the database before granting access to any table included in the database.
+
+The following example illustrates granting privileges on the following objects to the t share created in the previous step:
+
+```sql
+grant usage on database db1 to share t;
+
+grant select on table db1.table1 to share t;
```

Comment:
Is it possible to apply `usage` to a `table`?

If not, there should be a spec about what privilege can be applied to each kind of object.

Reply:
has added more comments on privilege.